toolbox/relnotes/main.go

```diff
@@ -0,0 +1,757 @@
+		createDownloadsTable(f, releaseTag, "Node Binaries", releaseTars+"/kubernetes-node*.tar.gz")
+		f.WriteString("\n")
+	}
+}
+
+// createDownloadTable creates table of download link and sha256 hash for given file.
+func createDownloadsTable(f *os.File, releaseTag, heading string, filename ...string) {
+	var urlPrefix string
+
+	if *releaseBucket == "kubernetes-release" {
+		urlPrefix = k8sReleaseURLPrefix
+	} else {
+		urlPrefix = fmt.Sprintf("https://storage.googleapis.com/%s/release", *releaseBucket)
+	}
+
+	if heading != "" {
+		f.WriteString(fmt.Sprintf("\n### %s\n", heading))
+	}
+
+	f.WriteString("\n")
+	f.WriteString("filename | sha256 hash\n")
+	f.WriteString("-------- | -----------\n")
+
+	files := make([]string, 0)
+	for _, name := range filename {
+		fs, _ := filepath.Glob(name)
+		for _, v := range fs {
+			files = append(files, v)
+		}
+	}
+
+	for _, file := range files {
+		fn := filepath.Base(file)
+		sha, err := u.GetSha256(file)
+		if err != nil {
+			log.Printf("failed to calc SHA256 of file %s: %s", file, err)
+			os.Exit(1)
+		}
+		f.WriteString(fmt.Sprintf("[%s](%s/%s/%s) | `%s`\n", fn, urlPrefix, releaseTag, fn, sha))
+	}
+}
+
+// minorReleases performs a minor (vX.Y.0) release by fetching the release template and aggregate
+// previous release in series.
+func minorRelease(f *os.File, release, draftURL, changelogURL string) {
+
+	// Check for draft and use it if available
+	log.Printf("Checking if draft release notes exist for %s...", release)
+
+	resp, err := http.Get(draftURL)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+
+	// TODO: find a better way to tell failed response
+	if err == nil && (resp.StatusCode == 200 || resp.StatusCode == 304) {
+		log.Printf("Draft found - using for release notes...")
+		_, err = io.Copy(f, resp.Body)
+		if err != nil {
+			log.Printf("error during copy to file: %s", err)
+			return
+		}
+		f.WriteString("\n")
+	} else {
+		log.Printf("No draft found - creating generic template...")
+		f.WriteString("## Major Themes\n\n* TBD\n\n## Other notable improvements\n\n* TBD\n\n## Known Issues\n\n* TBD\n\n## Provider-specific Notes\n\n* TBD\n\n")
+	}
+
+	// Aggregate all previous release in series
+	f.WriteString(fmt.Sprintf("### Previous Release Included in %s\n\n", release))
+
+	// Regexp Example:
+	// Assume the release tag is v1.7.0, this regexp matches "- [v1.7.0-" in
+	//     "- [v1.7.0-rc.1](#v170-rc1)"
+	//     "- [v1.7.0-beta.2](#v170-beta2)"
+	reAnchor, _ := regexp.Compile(fmt.Sprintf("- \\[%s-", release))
+
+	resp, err = http.Get(changelogURL)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+
+	// TODO: find a better way to tell failed response
+	if err == nil && (resp.StatusCode == 200 || resp.StatusCode == 304) {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(resp.Body)
+		for _, line := range strings.Split(buf.String(), "\n") {
+			if anchor := reAnchor.FindStringSubmatch(line); anchor != nil {
+				f.WriteString(line + "\n")
+			}
+		}
+		f.WriteString("\n")
+	}
```

Comment:
What if err != nil. Should we have some message out?

Reply:
Printed out error message/status code